pgrx/src/guc.rs

```diff
@@ -211,13 +212,133 @@ where
 /// A struct that has associated functions to register new GUCs
 pub struct GucRegistry {}
 impl GucRegistry {
+    // GUC Registration functions that do not expose hooks
     pub fn define_bool_guc(
         name: &str,
         short_description: &str,
         long_description: &str,
         setting: &GucSetting<bool>,
         context: GucContext,
         flags: GucFlags,
+    ) {
+        GucRegistry::define_bool_guc_with_hooks(
+            name,
+            short_description,
+            long_description,
+            setting,
+            context,
+            flags,
+            None,
+            None,
+            None,
+        );
+    }
+
+    pub fn define_int_guc(
+        name: &str,
+        short_description: &str,
+        long_description: &str,
+        setting: &GucSetting<i32>,
+        min_value: i32,
+        max_value: i32,
+        context: GucContext,
+        flags: GucFlags,
+    ) {
+        GucRegistry::define_int_guc_with_hooks(
+            name,
+            short_description,
+            long_description,
+            setting,
+            min_value,
+            max_value,
+            context,
+            flags,
+            None,
+            None,
+            None,
+        );
+    }
+
+    pub fn define_string_guc(
+        name: &str,
+        short_description: &str,
+        long_description: &str,
+        setting: &GucSetting<Option<&'static CStr>>,
+        context: GucContext,
+        flags: GucFlags,
+    ) {
+        GucRegistry::define_string_guc_with_hooks(
+            name,
+            short_description,
+            long_description,
+            setting,
+            context,
+            flags,
+            None,
+            None,
+            None,
+        );
+    }
+
+    pub fn define_float_guc(
+        name: &str,
+        short_description: &str,
+        long_description: &str,
+        setting: &GucSetting<f64>,
+        min_value: f64,
+        max_value: f64,
+        context: GucContext,
+        flags: GucFlags,
+    ) {
+        GucRegistry::define_float_guc_with_hooks(
+            name,
+            short_description,
+            long_description,
+            setting,
+            min_value,
+            max_value,
+            context,
+            flags,
+            None,
+            None,
+            None,
+        );
+    }
+
+    pub fn define_enum_guc<T>(
+        name: &str,
+        short_description: &str,
+        long_description: &str,
+        setting: &GucSetting<T>,
+        context: GucContext,
+        flags: GucFlags,
+    ) where
+        T: GucEnum<T> + Copy,
+    {
+        GucRegistry::define_enum_guc_with_hooks(
+            name,
+            short_description,
+            long_description,
+            setting,
+            context,
+            flags,
+            None,
+            None,
+            None,
+        );
+    }
+
+    // GUC Registration functions that expose hooks
+    pub fn define_bool_guc_with_hooks(
```

Comment:
```suggestion
    /// Define a boolean GUC with custom hooks.
    /// 
    /// # Hooks
    /// 
    /// * `check_hook` - Validates new values. Return false to reject.
    /// * `assign_hook` - Called after value is set. Use for side effects.
    /// * `show_hook` - Returns custom display string for SHOW commands.
    /// 
    /// # Safety
    /// 
    /// Hook functions must use `#[pg_guard]` and handle errors properly.
    pub fn define_bool_guc_with_hooks(
```

I would love to have something like this here, if you agree.